coap/api/transport.go

```diff
@@ -66,75 +66,78 @@ func MakeCoAPHandler(svc coap.Service, l *slog.Logger) mux.HandlerFunc {
 	return handler
 }
 
-func sendResp(w mux.ResponseWriter, resp *message.Message) {
-	if err := w.Client().WriteMessage(resp); err != nil {
+func sendResp(w mux.ResponseWriter, resp *pool.Message) {
+	if err := w.Conn().WriteMessage(resp); err != nil {
 		logger.Warn(fmt.Sprintf("Can't set response: %s", err))
 	}
 }
 
 func handler(w mux.ResponseWriter, m *mux.Message) {
-	resp := message.Message{
-		Code:    codes.Content,
-		Token:   m.Token,
-		Context: m.Context,
-		Options: make(message.Options, 0, 16),
+	resp := pool.NewMessage(w.Conn().Context())
+	resp.SetToken(m.Token())
+	for _, opt := range m.Options() {
+		resp.AddOptionBytes(opt.ID, opt.Value)
 	}
-	defer sendResp(w, &resp)
+	defer sendResp(w, resp)
+
 	msg, err := decodeMessage(m)
 	if err != nil {
 		logger.Warn(fmt.Sprintf("Error decoding message: %s", err))
-		resp.Code = codes.BadRequest
+		resp.SetCode(codes.BadRequest)
 		return
 	}
 	key, err := parseKey(m)
 	if err != nil {
 		logger.Warn(fmt.Sprintf("Error parsing auth: %s", err))
-		resp.Code = codes.Unauthorized
+		resp.SetCode(codes.Unauthorized)
 		return
 	}
-	switch m.Code {
+
+	switch m.Code() {
 	case codes.GET:
-		err = handleGet(m.Context, m, w.Client(), msg, key)
+		err = handleGet(m, w, msg, key)
```

Comment:
Se the status code then call handleGet
```suggestion
		resp.SetCode(codes.Content)
		err = handleGet(m, w, msg, key)
```